internal/server/routes.go

```diff
@@ -402,3 +438,21 @@ func (a *API) notFoundHandler(c *gin.Context) {
 		logging.Errorf("%s", err.Error())
 	}
 }
+
+func (a *API) deprecatedRoutes(noAuthnNoOrg *routeGroup) {
+	// CLI clients before v0.14.4 rely on sign-up being false to continue with login
+	type SignupEnabledResponse struct {
+		Enabled bool `json:"enabled"`
+	}
+
+	add(a, noAuthnNoOrg, http.MethodGet, "/api/signup", route[api.EmptyRequest, *SignupEnabledResponse]{
+		handler: func(c *gin.Context, _ *api.EmptyRequest) (*SignupEnabledResponse, error) {
+			return &SignupEnabledResponse{Enabled: false}, nil
+		},
+		routeSettings: routeSettings{
```

Comment:
Do we need a `defaultDeprecatedRouteSettingsGet` similar to `defaultRouteSettingsGet`? I would imagine as we add deprecated routes here we're going to pile up routeSettings.

Side note, the way we add routeSettings to the route feels weird given that we pass in part of the arguments through the settings here, but the `route.authenticationOptional` and `route.organiaztionOptional` settings get set through the route group (and have different names for the settings)

Reply:
> Do we need a `defaultDeprecatedRouteSettingsGet` similar to `defaultRouteSettingsGet`?

Maybe, I guess we'll see how many deprecated routes we end up with.

----

> the way we add routeSettings to the route feels weird given that we pass in part of the arguments through the settings here, but the route.authenticationOptional and route.organiaztionOptional settings get set through the route group (and have different names for the settings)

Fixing the names is easy enough.  The `routeGroup` struct has a TODO on it to be removed. I think we should get rid of route groups and set everything on package level `var` for each route. There are a couple routes that already do that.  That's going to be necessary for #3562.

I think that would fix the problem because all settings would come from that `var` . The `routeGroup` would no longer exist.

Reply:
I fixed the names of the `routeGroup` fields in this PR.